src/stylesheets/components/_modal.scss

```diff
@@ -0,0 +1,176 @@
+.usa-js-no-click {
+  pointer-events: none;
+  user-select: none;
+
+  .usa-modal,
+  .usa-modal * {
+    pointer-events: auto;
+    user-select: text;
+  }
+}
+
+// Prevents propagation issue in Safari an opener
+// has a child element, and when clicked, prevents
+// intended action
+[data-open-modal] * {
+  pointer-events: none;
+}
+
+.usa-modal-wrapper {
+  text-align: center;
+  transition: opacity $project-easing;
+
+  // Using visibility: hidden instead of display: none because
+  // of iOS VoiceOver issue:
+  // https://developer.paciellogroup.com/blog/2018/06/the-current-state-of-modal-dialog-accessibility/
+  &.is-hidden {
+    visibility: hidden;
+    opacity: opacity(0);
+    position: fixed;
+  }
+
+  &.is-visible {
+    visibility: visible;
+    opacity: opacity(100);
+    position: fixed;
+    z-index: z-index($theme-modal-z-index);
+  }
+}
+
+.usa-modal-overlay {
+  background: color("black-transparent-50");
+  bottom: 0;
+  height: 100%;
+  left: 0;
+  overflow: scroll;
+  overflow-x: hidden;
+  padding: units(3);
+  position: fixed;
+  scroll-behavior: smooth;
+  top: 0;
+  width: 100%;
+
+  &:before {
+    content: "";
+    display: inline-block;
+    height: 100%;
+    vertical-align: middle;
+  }
+
+  &[data-force-action="true"] {
+    pointer-events: none;
+
+    * {
+      pointer-events: auto;
+    }
+  }
+}
+
+// Prevents modals from flickering
+// Still allows content to be accessed
+.usa-js-loading {
+  .usa-modal-wrapper {
+    @include add-sr-only;
+  }
+
+  .usa-modal-wrapper:target {
+    position: static;
+  }
+}
+
+.usa-modal {
+  @include u-radius($theme-modal-border-radius);
+  @include u-shadow($theme-modal-box-shadow);
+  background: white;
+  display: inline-block;
+  margin: units(2.5) auto;
+  max-width: $theme-modal-default-max-width;
+  position: relative;
+  text-align: left;
+  vertical-align: middle;
+
+  &:focus {
+    outline: none;
+  }
+}
+
+.usa-modal__content {
+  display: flex;
+  flex-direction: column-reverse;
+  padding-top: units(4);
+  width: 100%;
+}
+
+.usa-modal__main {
+  margin: 0 auto;
+  padding: units(1) units(4) units(4);
+}
+
+.usa-modal__header {
```

Comment:
Is the intention that `usa-modal__heading` live inside here or is this separate?

Reply:
The way we usually use this is that a `header` is a section and a `heading` is an `h`-level element. We currently have a heading, but it looks like the header is not yet implemented. I'm going to remove this code